src/meshapi/widgets.py

```diff
@@ -0,0 +1,34 @@
+import json
+from django.forms import Widget
+from django.template import loader
+from django.utils.safestring import mark_safe
+from django_jsonform.widgets import JSONFormWidget
+
+
+class PanoramaViewer(JSONFormWidget):
+    pano_template_name = "widgets/panorama_viewer.html"
+
+    def __init__(self, schema):
+        super().__init__(schema)
+
+    def pano_get_context(self, name, value, attrs=None):
+        # FIXME: Need to parse the value because Django gives it as a string
+        # for some reason
+        value_as_array = json.loads(value)
+        return {
+            "widget": {
+                "name": name,
+                "value": value_as_array,
+            }
+        }
+
+    def render(self, name, value, attrs=None, renderer=None):
+        # Render the JSONFormWidget to allow editing of the panoramas
```

Comment:
What kind of editing to do we allow? How do we ensure that edits are not "blown away" by the next import

Reply:
Hmm, good point. I think I need to update the import logic to account for that. Can probably just check if the order/length is different from expected and instead of replacing the field, append it.

Reply:
Alternatively, Olivier is comfortable working with the current system, we could keep ours read-only until we can replace the GitHub

Reply:
I will add an env var to remove the edit field